src/components/ui/time-picker.svelte

```diff
@@ -0,0 +1,69 @@
+<script>
+  import { createEventDispatcher } from 'svelte';
+  import Button from './button.svelte';
+  import TextField from './text-field.svelte';
+
+  export let value = { hours: null, minutes: null };
+  if (value == null) {
+    value = { hours: null, minutes: null };
+  }
+  const minutesInHour = 60;
+  const hoursInDay = 24;
+
+  const dispatch = createEventDispatcher();
+
+  function setHours(hours) {
+    if (hours < 0) {
+      value.hours = hoursInDay - 1;
+    } else if (hours >= hoursInDay) {
+      value.hours = 0;
+    } else {
+      value.hours = hours;
+    }
```

Comment:
Since it seems like you don't like using modular arithmetic here, I think
```js
    hours = Math.min(hours, hoursInDay - 1);   // to prevent overflow
    hours = Math.max(hours, 0);   // to prevent underflow (negative)
    value.hours = hours;
```
looks more idiomatic.

Reply:
Idiomatic? Don't think so. Cleaner in terms of code aesthetic, maybe. Expressive, not really. In this case I'd opt for clarity instead of beauty